server/src/main/java/org/opensearch/action/get/TransportGetAction.java

```diff
@@ -88,16 +92,32 @@ protected boolean resolveIndex(GetRequest request) {
         return true;
     }
 
+    protected static boolean isSegmentReplicationEnabled(ClusterState state, String indexName) {
+        return Optional.ofNullable(state.getMetadata().index(indexName))
+            .map(
+                indexMetadata -> ReplicationType.parseString(indexMetadata.getSettings().get(IndexMetadata.SETTING_REPLICATION_TYPE))
+                    .equals(ReplicationType.SEGMENT)
+            )
+            .orElse(false);
+    }
+
+    /**
+     * Returns true if GET request should be routed to primary shards, else false.
+     */
+    protected boolean isPrimaryBasedRouting(ClusterState state, GetRequest request, String indexName) {
+        return request.realtime() && request.preference() == null && isSegmentReplicationEnabled(state, indexName);
```

Comment:
nit - for the vast majority of requests realtime will be true & preference will not be set - lets check the SR setting first to avoid the redundant checks?

Can we also make this static and take boolean realtime and preference as params so we can reuse this in mget?  Will also be easier to test.